### ISSUE TITLE
remacs-bindings generating bad source

### ISSUE THREAD

**Issue by enebo** [Wilfred/remacs#1199]
When running make:

```text
     Running `/home/enebo/work/remacs/rust_src/remacs-bindings/target/release/remacs-bindings bindings ../rust_src/generated/bindings.rs`
Rustfmt failed at stdin:185: line exceeded maximum length (maximum: 100, found: 193) (sorry)
Rustfmt failed at stdin:2869: line exceeded maximum length (maximum: 100, found: 364) (sorry)
Rustfmt failed at stdin:9638: line exceeded maximum length (maximum: 100, found: 108) (sorry)
Rustfmt failed at stdin:9956: line exceeded maximum length (maximum: 100, found: 247) (sorry)
.... many more
```

I believe this is a warning but then later on when it tries to compile the generated source:

```text
error: unexpected close delimiter: `]`
     --> src/../generated/bindings.rs:15591:32
      |
15591 | #[derive(Copy, Clone)]; 4usize ] , }#[repr(C)]
      |                                ^ unexpected close delimiter

error: aborting due to previous error
```

Not entirely sure what else I can profile at this point to.  I am at commit 2512475f3ff049fff93d951d4a4c14c93b1a583d.

**Comment by shaleh** [comment]
hmmm.

I just cloned the tree and did a build. I ran configure like this `CFLAGS='-O0 -g3' ./configure --enable-checking='yes,glyphs' --enable-check-lisp-object-type --enable-rust-debug`. This is a debuggable build configuration. The compilation went fine with no complaints.

Check your tree?

**Comment by enebo** [comment]
Note for posterity...I had a really old rustfmt binary in ~/.cargo/bin.  Once I removed that then the build worked.  I can only guess the preferences from remacs via rustup was not picking up the proper fmt but unconditionally using the old one in that directory?  

So this I think is not something which is remacs problem and I am not sure I can really provide helpful text here since I think this may be an obscure env issue I happened to hit by not updating rust in a terribly long time.
